internal/plugins/ansible/init.go

```diff
@@ -0,0 +1,115 @@
+/*
+TODO(asmacdo) sdk licence header
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ansible
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+	"sigs.k8s.io/kubebuilder/pkg/plugin"
+	"sigs.k8s.io/kubebuilder/pkg/plugin/scaffold"
+
+	"github.com/operator-framework/operator-sdk/internal/kubebuilder/cmdutil"
+	"github.com/operator-framework/operator-sdk/internal/plugins/ansible/scaffolds"
+)
+
+type initPlugin struct {
+	config    *config.Config
+	apiPlugin createAPIPlugin
+
+	generatePlaybook bool
+	doAPIScaffold    bool
+
+	// For help text.
+	commandName string
+}
+
+var (
+	_ plugin.Init        = &initPlugin{}
+	_ cmdutil.RunOptions = &initPlugin{}
+)
+
+// TODO(asmacdo) documentation
+func (p *initPlugin) UpdateContext(ctx *plugin.Context) {
+	p.commandName = ctx.CommandName
+}
+
+func (p *initPlugin) BindFlags(fs *pflag.FlagSet) {
+	fs.SortFlags = false
+	fs.StringVar(&p.config.Domain, "domain", "my.domain", "domain for groups")
+	p.apiPlugin.BindFlags(fs)
+	// TODO(asmacdo) short flags?
+	fs.BoolVarP(&p.generatePlaybook, "generate-playbook", "", false, "Generate a playbook skeleton. (Only used for TODO(asmacdo)--type ansible)")
```

Comment:
```suggestion
	fs.BoolVarP(&p.generatePlaybook, "generate-playbook", "", false, "Generate a playbook skeleton.")
```
Note required because when we run init --plugin=name it will show just the flags required for the specific plugin.

Reply:
I left it in `init` because it affects the Dockerfile. https://github.com/operator-framework/operator-sdk/pull/3488/files#diff-915c7657ba526a66ef1cf80ed660a892R84. IIRC, when I tried to access this value in scaffolds/init.go, it complained because `NewInitScaffolder` is a `scaffold.Scaffold`, which doesn't have this value.

Reply:
IMO the init create the Dockerfile and the api would update the Dockerfile when it is set. 
See an example over how you can update it [here](https://github.com/kubernetes-sigs/kubebuilder/blob/76edcfba51f581ea8c1f75f5cb8ba4c730f258ca/pkg/plugin/v2/scaffolds/edit.go#L53





)

Reply:
I've updated this, `init` creates the Dockerfile and `api` adds roles/playbooks as necessary.

Reply:
I've updated this again. Now, `init` creates a static Dockerfile and also scaffolds roles/ and playbooks/ dir, which can be filled in by `api`. This simplifies everything pretty well.